### DISCUSSION TITLE
Listing Ring,Kton,Crab,Ckton on DEX

### DISCUSSION THREAD

**Discussion by SasoLithops** [ktondao/KtonDAO#28] [Q&A]
Hi!

My question is does Darwinia have any thoughts of making native Crab, Ckton, Ring and Kton available on any DEX? It is now only on Snowswap and liquidity is low for Kton and none for Ckton. Which is really bad for token holders/buyers/sellers. There is no price check for Crab and Ckton as well except pairing on Snowswap with usdt. Especialy impossible for Ckton. Liquidity  providing is left for users although one needs to be very active, smart, experienced whale and have knowledge of IL to provide it as from users liquidity can only be as deep. I think many got burned hard already. I did too the first time although incentives then covered most of my IL. Buying Crab is nightmare to regular members(been helping a few in private chat, they think it is highly complicated). 
I have been checking many DEXs in Dotsama space with many of them we have open channels as parachain so please let me know if I can help. I am sure Jesse can help too. And some other community members as well.
This question is not for my personal interest alone but comes up many times from community members.
So we can start discussion if no option available yet on how to approach this. I don't want Darwinia to provide all the liquidity and be left out dry but if we could somehow do it together.
Please let me know what you all think could help to get this problem fixed.

**Comment by hackfisher** [comment]
I think we need to first clarify and understand the question, especially if there are some conceptual confusions. Darwinia, in itself, should refer only to this network or community, which involves many entities or community members, such as Darwinia Treasury, Itering, DCDAO, Crab DAO, HelixDAO, SubAPI DAO, Councils, ambassadors, independent developers, app developers, token holders, and so on. Therefore, asking 'Does Darwinia have any thoughts?' sounds a bit strange and hard to answer because “There might be many thoughts in the Darwinia Community from different people and entities.” Asking specific organizations or individuals would make it easier to answer.

For DEX listings, in my opinion, most of them are permissionless, which is also the spirit and characteristic of Web3. If someone wants to list, they can go directly and add liquidity."

If you have already created a trading pair on a DEX, please share the addresses for trading and adding liquidity pools (LP). This allows other members of the community to contribute to the liquidity as well.

If you think confirming more documents and community resources is necessary to help beginners learn how to use DEX and get ring, please seek assistance from support and education personnel within the community.

If you are unsure about how to create a DEX and need technical help, please provide more detailed descriptions with additional details.

Thank you.

**Reply by SasoLithops** [reply]
Hi!

Thank you for your reply.

I am used to writing on TG and Discord where by Darwinia we have always meant what is now Darwinia Github Organization(DGO). I appologize for my mistake, will be more careful in future if this brought confusion. I definitely want this disscusion to be for every Darwinia Community member. However I am firstly and primarily interested specificaly in DGO opinion first. Will elaborate on why.

I didn't create trading pair but participated in I think it was xRING/USDT and CRAB/USDC on Snowswap back in April 2022 when Darwinia(Council) created incentivised liquidity pools. I attach 

https://medium.com/darwinianetwork/darwinia-will-offer-1-000-000-xring-and-7-000-000-crab-for-usdt-xring-and-usdc-crab-incentives-for-31ecec7fe843

Maybe I wasn't clear enough. I have no intention to launch DEX:-) My question is:

Does DGO THINK about providing some RING, KTON, CRAB and CKTON liquidity on some Polkadot DEXS? 

Hope it makes sense now. I just want opinion from DGO before acting further. 

Currently the only way to buy Crab is on Snowswap and liquidity there with let's say still accebtable 5.72% price impact in xWring/Crab pool alows you to only spend 200.000 Ring and buy cca 3.000.000 Crab. So conisdering this Crab is impossible to get in larger quantities.

Ckton/Crab pool is drained out so we CKTON holders can not buy it any more or are left stuck with it(I have my Crab and Ring locked for couple years so I am not looking for escape liquidity)

This imho gives opinion that no one cares about Crab, CKTON holders and that is why I think we must act on this. Counting on community members to provide liquidity on their own if they want is I think too much to ask I explained in first post why.

Hope now my point makes more sense.

I would also like to know if it is possible to get data on how much yearly traffic has been in 2023 on DEXs and CEXs trading Ring, Kton, Crab and Ckton? Some CEXs listing Ring have become very shady as well. I think much broader debate needs to be set regarding having good options on getting all Darwinia tokens.

**Reply by pdmerritt** [reply]
I can't speak on crab or ckton but I would definitely like to see ring and kton on a dex. I would really like to see it on the HydraDX Omnipool I know this would have to be a community decision but i'm just putting it out there. I am aware that there are ways to basically list a token without the community but that would mean you can provide liquidity yourself. I'm not in the position to do this so the treasury would have to be involved or a bunch of, like minded, individuals with deep pockets to provide the liquidity required. 
This would make for much easier on an off ramps for everyone in the community as well as more cost effective. 

I even reached and asked what the process is and this was the info provided which includes an example from kilt...everything conveyed to me is here:

Jose Crypto, [1/19/2024 1:39 PM]
Rn the process is, need to have a HRMP channel with the parachain origin 

Need to register the asset on our chain

Also need to agree who will provide the initial liquidity  ( usually do an ref proposal on their chain with all the info

Similar to this one with kilt for example
https://kilt.polkassembly.io/post/233)

We also do a ref on our side to pre agree on fas track the add to Omnipool ref

Jose Crypto, [1/19/2024 1:41 PM]
Also since all tokens are correlated in the Omnipool 

We should only put tokens that had open governance controlled by holders ( not sudo or something similar), most of the tokens already on circ

And also need to have price discovery, ie cant be launched on Omnipool, need to be traded of other way befofe

Jose Crypto, [1/19/2024 1:41 PM]
Also on HydraDX we had LBP and isolated pool for launchs and price discovery of tokens

Jose Crypto, [1/19/2024 1:42 PM]
But not are in the Omnipool 

And example is the SUB/DOT isolated pool

Jose Crypto, [1/19/2024 1:42 PM]
Is routed via Omnipool with DOT of course, but is not in correlation with the other assets in the Omnipool

**Reply by hackfisher** [reply]
My favourite DEX for RING is Degate: https://app.degate.com/trade/USDC/0x9469d013805bffb7d3debe5e7839237e535ec483

Usually, DEX like Uniswap and Degate will help provide data analysis for the trading pairs, there are other data platforms like coingeco will aggregate the data, for example we can compare the depth on coingeco for now.

There are two direction need to strengthen so the community and gov can be guided to add liquidity or using them.
1. Integration and docs with these data platform and DEX, so anyone with exposure to the tokens you mentioned can easily add liquidity and trade on them.
2. Better data analysis for the community, so the data driven view can guide the gov system and community to decide the long term strategy.

Both of them are covered and initiated by the [DCDAO Ecosystem Program](https://github.com/orgs/dcdao/discussions/55), join as Integration Specialist and Data Analyst if you are interested!

For adding liquidity and trade, why not join and get started with Snowswap, Uniswap and Degate now? Or do you think it is hard for community members to do these things, and need more education and support. (I proposed to add a "Get RING" page in next version of the website, do you think that's enough?)

**Reply by hackfisher** [reply]
For  HydraDX Omnipool, that's a good initiative, @AurevoirXavier would you please take a look at this and provide necessary technical support as this is a proposal in Polkadot ecosystem?

**Reply by aurexav** [reply]
Okay, I've added it to my list.

**Reply by SasoLithops** [reply]
It is my first time hearing for Degate. Who is providing liquidity there and how come Ring is traded with USDC? Is that on Ethereum network, I can not check as not at home.

I still think adding liquidity in Snowswap is too hard for community members alone. Many fear that Impermenent loss will eat most of their profits and keep to staking(they are right too) as pool is usually not deep enough. If it wasn't for Dustin who knows how to provide liquidity and how/when to exit from LP most probably none would dare to provide and risk their tokens. He since removed liquidity due to reasons he named (sorry I forgot)and he shared all of this with community when he did which is highly appreciated and transparent. Many community members did try to provide liquidity too but with small amounts that really doesn't help the pool as swapping only a few tokens already made a huge price impact.   

I think it is left up to a project to first put some ammount for initial or permanent liquidity on DEX as this puts more trust for community to join. Maybe some incentives for providing liquidity could also be another option to cover some of the IL community members take on(sometimes DEXes also offer some of their tokens if they have them as well for couple of months) I have seen this being the case with projects similar to ours. 

What do you think? Is Darwinia governance ready for proposal to provide some initial liquidity in DEX or incentivise people to do so?

I would join if I had more free time and knowledge. I lack in both departments;-)

**Reply by Laki389** [reply]
> For adding liquidity and trade, why not join and get started with Snowswap, Uniswap and Degate now? Or do you think it is hard for community members to do these things, and need more education and support. (I proposed to add a "Get RING" page in next version of the website, do you think that's enough?)

@hackfisher 

The problem with the lack of liquidity on Snowswap (or any Polkadot ecosystem DEX, etc.,) is not due to lack of education, support or similar, it has been written many times, but it is worth repeating if it helps to overcome the problem, the lack of liquidity for native Kton and Ring trading.

Community members don't put Kton and Ring as liquidity because of staking and not having how to attract and with what to attract to do it and have a bigger benefit than simply staking their Kton and Ring.

I see a solution only in the KTON that is in Darwinia Treasury (or whatever that wallet is officially called) and in another, how should I call that wallet, that is not owned by community members, and in which there are also some 300+ KTON, a total of 700 maybe KTON and which would be great liquidity.

**Reply by Laki389** [reply]
I found the wallets,

- https://darwinia.subscan.io/account/0x6d6f646c64612f74727372790000000000000000   **477 KTON**

- https://darwinia.subscan.io/account/0xb5338bf727049740f886a33b7f0d2b1a291fea22  **302 KTON**

**Comment by wuminzhe** [comment]
DCDAO is recruiting people to do the work.

**Reply by SasoLithops** [reply]
May I know what work?

**Reply by wuminzhe** [reply]
@SasoLithops I mean [this discussion](https://github.com/orgs/dcdao/discussions/55)

**Comment by pdmerritt** [comment]
Ring and Kton are EVM tokens. It seems like they may be trying to come up
with their own solution from crab and ckton.

**Comment by Laki389** [comment]
> If it wasn't for Dustin who knows how to provide liquidity and how/when to exit from LP most probably none would dare to provide and risk their tokens. 

@SasoLithops 
It's not about knowing when to put liquidity, but about conscious sacrifice and giving up staking rewards, but when for a long time only 1 person provides 90 or 95% of the liquidity, it still leaves a bitter feeling.

So there is currently no liquidity for KTON on Snowswap, nor is there any way to buy or sell Ring.

As a result of this and other reasons, Darwinia does not have more than a dozen transfers per day.

**Reply by SasoLithops** [reply]
But Laki remember when we were having staking discussion and many got loud we are not receiving less rewards then we used to. We want high percentage to go to stakers. And not giving higher percentage to treasury. 
If we sacrificed some of rewards put more percentage to the treasury, gave some of that percentage of that treasury to liquidity providers as incentives or simply buy back Kton, we wouldn't have this conversation. We have too attractive staking rewards. My power dropped 2M+ and I still receive more then I used to. We only " incentivise" stakers.

What if we meet in half way. Have a "halving" in staking rewards(so staking is not that atractive, many others have done it)  and put the other half in treasury to incentivise liquidity providers and  dApps to start building on our chain.

Other option is restaking will be sooner or later comming to Polkadot and with our huge staking rewards protocols might start using it. I need to dig deeper to see if that could be profitable for us.

**Reply by Laki389** [reply]
> But Laki remember when we were having staking discussion and many got loud we are not receiving less rewards then we used to. We want high percentage to go to stakers. And not giving higher percentage to treasury. If we sacrificed some of rewards put more percentage to the treasury, gave some of that percentage of that treasury to liquidity providers as incentives or simply buy back Kton, we wouldn't have this conversation. We have too attractive staking rewards. My power dropped 2M+ and I still receive more then I used to. We only " incentivise" stakers.
> 

I think we have discussed this many times and there is no point in going around in circles, but let's do it for the last time.. do you know another project from the Polkadot ecosystem where 60% of the annual inflation goes to the Treasury, tell me 1 (one) other project ?

I think you are not informed, or you are consciously or unconsciously manipulating.

I found my calculation from November 27, 2023, according to which with only 14% of the entire Ring being staked, staking rewards are already 12% APR, which means that if there was a 50% staking rate as it is approximately on Polkadot there would be an absurd situation where the APR would be less than the annual inflation.

After 2 months from then, the situation can only worsen and for a certain period we will come to a situation where staking rewards are lower than inflation.

That's why I don't see your "attractive" staking rewards, lol.

What is it about, have you selected a collator of 15M total stake? Well, not everyone can do that, if everyone does that collator will have 800M total stake power.

Look for the middle, choose a collator of 40-45M total power stake and see your attractive rewards haha.

This is an average, you can also choose someone with 95M total stake, that's the other end of the story.

> What if we meet in half way. Have a "halving" in staking rewards(so staking is not that atractive, many others have done it) and put the other half in treasury to incentivise liquidity providers and dApps to start building on our chain.
> 
> Other option is restaking will be sooner or later comming to Polkadot and with our huge staking rewards protocols might start using it. I need to dig deeper to see if that could be profitable for us.

My suggestion is, these 2 wallets have 779 KTON in total and that is solid liquidity.

- https://darwinia.subscan.io/account/0x6d6f646c64612f74727372790000000000000000 **477 KTON**

- https://darwinia.subscan.io/account/0xb5338bf727049740f886a33b7f0d2b1a291fea22 **302 KTON**

**Reply by SasoLithops** [reply]
1. Let's not get personal I am not manipulating just expressing my opinion.
2. Hope this is last time too, I too have no more energy to talk endlessly the same thing. So why do you constantly think treasury is not ours? Wouldn't incentivising liquidity providers be given as rewards to community LP providers. Your callculations don't have that factor. They give all only to stakers. You constantly callculate with one variable. What if taking into account LP providers and potential price rise/fall and ... Treasury can be used in good ways to help the project grow which potentialy impacts price and "gives reward" to stakers too. Just not in short term.
3. My validator has 42M I can post my address here for you to check if you like and I look at it once a week. I am not that active at all.
4. Who are we to follow footsteps of every Polkadot project? Why not think out of the box? Name one in polkadot eco that is succsessful atm? https://www.coingecko.com/en/categories/dot-ecosystem
Other ecos call it deadchain.
5. Lets return to DEX debate and liquidity.

**Reply by Laki389** [reply]
> 1. Let's not get personal I am not manipulating just expressing my opinion.
> 
>     2. Hope this is last time too, I too have no more energy to talk endlessly the same thing. So why do you constantly think treasury is not ours? Wouldn't incentivising liquidity providers be given as rewards to community LP providers. Your callculations don't have that factor. They give all only to stakers. You constantly callculate with one variable. What if taking into account LP providers and potential price rise/fall and ... Treasury can be used in good ways to help the project grow which potentialy impacts price and "gives reward" to stakers too. Just not in short term.
> 
>     3. My validator has 42M I can post my address here for you to check if you like and I look at it once a week. I am not that active at all.
> 
>     4. Who are we to follow footsteps of every Polkadot project? Why not think out of the box? Name one in polkadot eco that is succsessful atm? https://www.coingecko.com/en/categories/dot-ecosystem
>        Other ecos call it deadchain.
> 
>     5. Lets return to DEX debate and liquidity.

I will not reply to you on this topic anymore, everyone has the right to their own opinion, and you have the right to your own opinion, have a nice day.


And now I'm writing in general, it doesn't apply to you, nor am I looking for your reply or opinion on my proposal,

I suggest to someone who has the desire, motivation, persistence and perseverance to push the KTON from the following 2 wallets, to be used for liquidity.

Since Darwinia, at least on paper, is decentralized, that KTON belongs to all of us, currently sitting there unused, so let it be used to provide liquidity to Snowswap, or perhaps another DEX, as the community decides.

- https://darwinia.subscan.io/account/0x6d6f646c64612f74727372790000000000000000 **477 KTON**
- https://darwinia.subscan.io/account/0xb5338bf727049740f886a33b7f0d2b1a291fea22 302 KTON

**Comment by pdmerritt** [comment]
I can't speak for the whole community but imho staking has been really good
to us for almost two years. If the project can be better by reducing the
staking rewards then I have no issue with it.
But to get back to the dex...Personally, I don't want to have to touch eth
if I don't have to because of the stupidly high fees! I'm sure there were
reasons for making ring and kton EVM compatible but I personally wish you
hadn't. I'm certain the entire community would enjoy simpler and less
expensive on and off ramps for their ring and kton though. It also makes it
more inviting for people onboarding for the first time. I'm sure there will
be new folks once the market is a lot more bullish. There also needs to be
a little more communication so people feel like the wheels are spinning.
Try to be more engaged with the community and it will go a long way.

**Reply by SasoLithops** [reply]
@pdmerritt I absolutely also speak only my opinion. We talk it out and see if it makes sense. I agree we should avoid ETH by any means exactlly because of high fees.
I also agree with communication that needs to improve. Maybe this is one step in right direction.

**Comment by SasoLithops** [comment]
To add to @PartnershipDude.

I was thinking to not change our staking model, we could provide incentives for farming liquidity on DEX. If treasury gave up 5% of its rewards, we would stop worrying all together about having  Kton listed somewhere and having it on DEX is imho enough for KTON.

Here are calculations behind this decision:

**418362 new Ring are minted daily** (thanks Laki and Jesse for calculating that number)

**NOW**
_60% of those go to treasury which comes daily to 251017.2 RING_

**AFTER**
If community thinks we could let go of 5% of this 60% which would now mean:

_55% still go to treasury which comes daily to 230099,1 RING
5% go to liquidity providers 20918,1 RING_

This number 5% is up to discussion I took into account that probably not many will provide liquidity soon after launch.  It would be good to get data how many farmed RING/USDT and CRAB/USDC liquidity when it was incentivised back in the days of Snowswap.

Maybe we could set this for 6 months and then vote on extending this if this proves to be effective.

**Comment by pdmerritt** [comment]
Also, if you're going to use a dex i'd really like it to be in the polkadot
ecosystem. Polkadot project NOT using the Polkadot ecosystem isn't
something I'd want to continue to support. What happened with The Omnipool?
I know they had been approached but it went silent afterwards.

**Reply by 2xJDubs** [reply]
Snowswap.xyz is the Darwinia native DEX so this would be farming on chain.

As far as I know the omnipool listing is moving forward with hrmp channels being opened as the first step.

**Comment by pdmerritt** [comment]
I was referring to the degate dex that had been mentioned.... Good news on
Omnipool though.

**Comment by sonoferin2000** [comment]
i know this is an old topic, but think it is still a valid one. we are in a market suge, but still no movement for crab, due to the world being blind to its existence. snowswap allows us to trade back and forth among ourselves, but it does not broadcast trade data to Coinmarketcap or Coinghecko, which has been really hurting us. what about possibly sending crab tokens across to Moonriver? could start an LP on solarbeam. or some other network where we have exposure to a more broader crypto market.,with good liquidity,and more eyeballs. its getting none through snowswap alone.. I think the team should select the best chain and DEX, and establish the  initial liquidity. .we should be able to see price feeds on CoinGecko. I think this would be good for crab to get eyeballs on it, as right now ,it isn't listed on coinmarketcap or coingecko. We need to get it listed on an exchange that gives us exposure to being included  in the price feeds. in the meantime, was wondering if we could get one of those dune analytics pages made up for crab and ckton, would be really helpful.there is no charting or graphs , we are totally in the dark.
